docs/configuring.md

```diff
@@ -0,0 +1,241 @@
+
+| Large page size  | System architecture required              | `-Xlp:codecache` | `-Xlp:objectheap`          |
+|------------------|-------------------------------------------|------------------|----------------------------|
+| 2 GB nonpageable | IBM zEnterprise EC12 processor or later   | Not supported    | Supported (64-bit VM only) |
+| 1 MB nonpageable | System z10 processor or later             | Not supported    | Supported (64-bit VM only) |
+| 1 MB pageable    | IBM zEnterprise EC12 processor or later (see Note) |Supported| Supported                  |
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** The Flash Express feature (#0402) helps avoids demoting 1 MB pageable pages to 4 KB pages during periods of high memory usage.
+
+If a particular page size cannot be allocated, a smaller page size is attempted, in descending order. For example, if 2 GB nonpageable pages are requested but not available, the VM tries to allocate 1MB nonpageable pages. If 1 MB nonpageable pages are not available, the VM tries to allocate 1MB pageable pages. If large pages are not available, 4 KB pages are allocated.
+
+If you want to use nonpageable large pages for the object heap, a system programmer must configure z/OS for nonpageable large pages in the **IEASYxx** parmlib member. Users who require large pages must also be authorized to the **IARRSM.LRGPAGES** resource in the RACF FACILITY class with read authority.
+
+Use the following z/OS system command to show large page usage for an LPAR:
+
+```
+DISPLAY VS,LFAREA
+```
+For more information, see [LFAREA](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.4.0/com.ibm.zos.v2r4.ieae200/lfareaxx.htm) in the z/OS product documentation.
+
+For usage information, including examples, see [`-Xlp:objectheap`](xlpobjectheap.md#pageablenonpageable).
+
+
+## Configuring Dynamic LPAR support (AIX only)
+
+*Dynamic logical partitioning* (DLPAR) provides a mechanism to add or remove system resources, such as memory or CPU, to or from the operating system in a logical partition without rebooting. Changing these resources dynamically can have an impact on Java applications that are running on the LPAR.
+
+To enable an application to respond to DLPAR events, you can use OpenJ9 extensions to the `java.lang.management` API. These extensions can listen for events and trigger any necessary adjustments to the runtime environment. For example, if a Java VM is running in an LPAR with 2GB of memory, but the available memory might be adjusted between 1GB and 8GB, you might set the following options for the Java heap at run time:
+
+```
+–Xms1g –Xsoftmx2g –Xmx8g
+```  
+
+This command-line string sets an initial heap size of 1 GB, a *soft* (adjustable) maximum heap size of 2 GB, and a maximum heap size of 8 GB. You can then use the API to adjust the size of [`-Xsoftmx`](xsoftmx.md) dynamically in response to changes in memory resources.  
```

Comment:
softmx support is unrelated to AIX DLPAR support, and is supported across platforms. For softmx, pls refer to MemoryMXBean getMaxHeapSize(), isSetMaxHeapSizeSupported(), setMaxHeapSize().

Reply:
the question is whether JVM is notified of memory resource changes without DLPAR enabled.  i guessed JVM can keep polling the OS to query the available memory resources.

Reply:
I believe softmx should not be mentioned in this context as an option for dynamic object heap size control. And yes, I agree it should be referred to MemoryMXBean

Reply:
Thanks .... clearer. I've modified:

>You can then use the `MemoryMXBean` API to dynamically respond to changes in memory resources. The following classes can be used:

- `getMaxHeapSize()`: Query the maximum heap size.
- `isSetMaxHeapSizeSupported()`: Query whether the VM can support dynamic updates.
- `setMaxHeapSize()`: Adjust the maximum heap size.

>For more information about the `com.ibm.lang.managment` package, which extends the `jdk.management` module, see the [API documentation](api-overview.md#monitoring-and-management).

Reply:
The softmx doc is still in the wrong place. It doesn't belong
a) in Configuring your system, since it's a JVM feature unrelated to the OS
b) under Configuring Dynamic LPAR support (AIX only), it's not an AIX specific feature and it's not related to DLPAR

Reply:
It's explaining a scenario where you might want to change the settings dynamically. i.e. For example, ....... you've created an environment with limits x,y,z ... and then when notified that things have changed, you can react. I accept it is not an OS feature or related to DLPAR. I'll remove if you think it confuses people .... or maybe I'm still misunderstanding your point.

Reply:
Ok, I see the point.